### PR TITLE
MAINT: linalg: use symbol suffix in fallback lapack_lite

### PR DESCRIPTION
**Pull request by pv**
When building the fallback lapack_lite library for numpy.linalg,
add symbol suffix to the routines when using 64-bit integer indices.

Adjust the lapack_lite generator script to also output a header file
that renames the symbols.  Moreover, also rename symbols from f2c.c, to
be sure (because they have short and generic names that may clash).

This avoids a name clash with BLAS/LAPACK names, which can cause
problems in corner cases on platforms where inadvertent symbol
overriding in dynamic libraries can occur (ie. on Linux, BLAS in global
symbol scope, from RTLD_GLOBAL or linked to program where Python
is embedded in).

### CODE REVIEW

File: numpy/linalg/lapack_lite/make_lite.py [hunk 1]
```diff
@@ -281,6 +285,44 @@ def ensure_executable(name):
...
+    # f2c symbols
+    f2c_symbols = set()
+    with open('f2c.h', 'r') as f:
+        for line in f:
+            m = extern_re.match(line)
+            if m:
+                f2c_symbols.add(m.group(1))
+
+    with open(os.path.join(output_dir, 'lapack_lite_names.h'), 'w') as f:
+        f.write(HEADER_BLURB + "\n")
```
Comment (eric-wieser) [inline numpy/linalg/lapack_lite/make_lite.py]: Mind adding a brief comment about what this file is for to the top of the file, just below this blurb?

Comment (pv) [reply numpy/linalg/lapack_lite/make_lite.py]: Added.

File: numpy/linalg/lapack_lite/make_lite.py [hunk 2]
```diff
@@ -281,6 +285,50 @@ def ensure_executable(name):
...
+            " * This file renames all BLAS/LAPACK and f2c symbols to avoid\n"
+            " * dynamic symbol name conflicts, in cases where e.g.\n"
+            " * integer sizes do not match with 'standard' ABI.\n"
+            " */\n")
+
+        # Rename BLAS/LAPACK symbols
+        for name in sorted(symbols):
+            f.write("#define %s_ BLAS_FUNC(%s)\n" % (name, name))
+
+        # Rename also symbols that f2c exports itself
```
Comment (eric-wieser) [inline numpy/linalg/lapack_lite/make_lite.py]: Perhaps
```suggestion
        # Rename also symbols that f2c exports itself
        f.write("\n"
                "/* Symbols exported by f2c.c */\n")
```

File: numpy/linalg/lapack_lite/python_xerbla.c [hunk 3]
```diff
@@ -2,9 +2,6 @@
 #include "numpy/npy_common.h"
 #include "npy_cblas.h"
 
-#undef c_abs
-#include "f2c.h"
```
Comment (eric-wieser) [inline numpy/linalg/lapack_lite/python_xerbla.c]: Not included simply because it's not needed?

File: numpy/linalg/setup.py [hunk 4]
```diff
@@ -41,7 +41,10 @@ def calc_info(self):
                 info = {'language': 'c'}
                 if sys.maxsize > 2**32:
                     # Build lapack-lite in 64-bit integer mode
-                    info['define_macros'] = [('HAVE_BLAS_ILP64', None)]
+                    info['define_macros'] = [
+                        ('HAVE_BLAS_ILP64', None),
+                        ('BLAS_SYMBOL_SUFFIX', '64_')
```
Comment (eric-wieser) [inline numpy/linalg/setup.py]: Is this chosen arbitrary, or does it match something else? If that latter, can we add a comment here?

File: numpy/linalg/lapack_lite/python_xerbla.c (hunk 3 above)
Comment (pv) [reply numpy/linalg/lapack_lite/python_xerbla.c]: Yes. The symbol rename also conflict here with BLAS_FUNC. This file is independent of lapack_lite, as it is also used when using a different blas.

**Comment by mattip** [comment]
@pv there are some comments waiting for replies

File: numpy/linalg/lapack_lite/make_lite.py (hunk 2 above)
Comment (pv) [reply numpy/linalg/lapack_lite/make_lite.py]: Done.

File: numpy/linalg/setup.py (hunk 4 above)
Comment (pv) [reply numpy/linalg/setup.py]: It's arbitrary. Added comment.

**Review by eric-wieser — APPROVED** [review]
Thanks for the explanatory comments @pv, to reiterate I'm shocked about how easy a fix this seemingly was.

**Comment by mattip** [comment]
Thanks @pv

**Pull request merged**